sdk/nodejs/runtime/closure/parseFunction.ts

```diff
@@ -234,21 +194,6 @@ function parseFunctionCode(funcString: string): [string, ParsedFunctionCode] {
         return makeFunctionDeclaration(constructorCode, /*isAsync:*/ false, /*isFunctionDeclaration: */ false);
     }
 
-    // Note: this check is pretty weak.  It is only checking if `=>` appears some place within the
-    // part of the code we consider the signature.  It's possible (though unlikely) for this check
-    // to be wrong.  For example, if the user wrote:
-    //
-    //      function F() /* a => b */ { }
-    //
-    // Then this would be treated as an arrow-function, even though it's a normal function.  However
-    // the likelihood of a comment including `=>` *and* preceding the opening `{` is low enough that
-    // we accept this as a good-enough heuristic.
-    const signature = funcString.substr(0, openCurlyIndex);
-    if (signature.indexOf("=>") >= 0) {
-        // (...) => { ... }
-        return ["", { funcExprWithoutName: funcString, isArrowFunction: true }];
-    }
```

Comment:
this code didn't work for cases like `({ de, con, struction }) => ...`   We always presumed that the `{` was the start of the block.